jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -422,7 +422,7 @@ ecma_copy_ecma_string (ecma_string_t *string_desc_p) /**< string descriptor */
       ecma_string_heap_header_t *new_data_p = (ecma_string_heap_header_t *) mem_heap_alloc_block (data_size);
       memcpy (new_data_p, data_p, data_p->size + sizeof (ecma_string_heap_header_t));
 
-      ECMA_SET_NON_NULL_POINTER (new_str_p->u.collection_cp, data_p);
+      ECMA_SET_NON_NULL_POINTER (new_str_p->u.collection_cp, new_data_p);
```

Comment:
I'm not sure that it is possible to construct a regression test case that can catch the refcount issue... It seems to me that this will be "just" eating up memory but not cause any semantic error. I'd be glad to be proven wrong though.

(The pointer issue is different. That might be possible to catch.)